lmsweb/views.py

```diff
@@ -104,6 +104,30 @@ def upload():
     # TODO: Extract the right exercise from the notebook
     #       (ask Efrat for code)
     # TODO: Check max filesize of (max notebook size + 20%)
+    exercise = Exercise.get_by_id(request.form.get('exercise', 0))
+    if not exercise:
+        # handle error
+        raise ValueError("exercise does not exist")
+    user = User.get_by_id(request.form.get('user', 0))
```

Comment:
We should check it against the session to see that this is the actual user.

This is a must to prevent form manipulation.

Reply:
Done by me.